### PR TITLE
Enable xa-transaction

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

## Checklist
- [ ] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Checked native-image compatibility

### CODE REVIEW

**Comment by daneshk** [comment]
@kalaiyarasiganeshalingam don't we need an additional docker image with XA transaction enabled for testing?

File: ballerina/Client.bal [hunk 1]
```diff
@@ -146,11 +146,13 @@ type ClientConfiguration record {|
 # + queryTimeout - Timeout (in seconds) to be used when executing a query.
 #                  (-1/0 means no query timeout)
 # + loginTimeout - Timeout (in seconds) to be used when connecting to the MSSQL server and authentication (default is 15s).
+# + useXADatasource - Flag to enable or disable XADatasource
 public type Options record {|
     SecureSocket secureSocket?;
     decimal socketTimeout?;
     decimal queryTimeout?;
     decimal loginTimeout?;
+    boolean useXADatasource = false;
```
Comment (niveathika) [inline ballerina/Client.bal:155]: Where is this processed?

**Review by niveathika — CHANGES_REQUESTED** [review]
`useXADatasource` is not processed

File: ballerina/tests/resources/Dockerfile [hunk 2]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM mcr.microsoft.com/mssql/server:2019-latest
+FROM mcr.microsoft.com/mssql/server:2017-latest
```
Comment (daneshk) [inline ballerina/tests/resources/Dockerfile:15]: why we downgrade the server?

**Pull request closed without merging**